protocol/grpc/client.go

```diff
@@ -25,15 +25,59 @@ import (
 	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
 	"github.com/opentracing/opentracing-go"
 	"google.golang.org/grpc"
+	"gopkg.in/yaml.v2"
 )
 
 import (
 	"github.com/apache/dubbo-go/common"
 	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/logger"
 	"github.com/apache/dubbo-go/config"
 )
 
-// Client ...
+var (
+	clientConf *ClientConfig
+)
+
+func init() {
+	// load clientconfig from consumer_config
+	// default use grpc
+	defaultClientConfig := GetDefaultClientConfig()
+	clientConf = &defaultClientConfig
```

Comment:
the above two line codes is very strange. why now define a `NewDefaultClientConfig() *ClientConfig`?

Reply:
ok,use defer delay check instead it